controller/src/test/java/io/pravega/controller/server/rpc/auth/ControllerGrpcAuthFocusedTest.java

```diff
@@ -573,4 +614,50 @@ private static File createAuthFile() {
         private final static String SCOPE1_STREAM2_READ = "authSc1Str2readonly";
         private final static String SCOPE1_STREAM1_LIST_READ = "scope1stream1lr";
     }
+
+    static class ResultObserver<T> implements StreamObserver<T> {
+        private T result = null;
+        private Throwable error;
+        private final AtomicBoolean completed = new AtomicBoolean(false);
```

Comment:
You only update this inside a `synchronized` block. Either remove the `synchronized` or make this a simple boolean.

Reply:
What about making it a future and completing the future upon the call to `onCompleted`? This way we can block on the call to `get()` (Line 645), which would simplify the implementation of that method and of others as the `this.wait` call is no longer necessary.

Reply:
I liked the idea of using the future for the purpose and have modified the code to use it instead.  That change addresses both the comments. Thanks for the suggestion - the code looks much more simpler now!